DigitalLearningSolutions.Data.Tests/Services/SessionServiceTests.cs

```diff
@@ -1,152 +1,99 @@
+    using FakeItEasy;
     using FluentAssertions;
+    using Microsoft.AspNetCore.Http;
     using NUnit.Framework;
 
-    internal class SessionServiceTests
+    public class SessionServiceTests
     {
-        private SessionService sessionService;
-        private SessionTestHelper sessionTestHelper;
+        private ISessionService sessionService;
+        private ISessionDataService sessionDataService;
+        private ISession httpContextSession;
+
+        private const int CandidateId = 11;
+        private const int CustomisationId = 12;
+        private const int DefaultSessionId = 13;
 
         [SetUp]
-        public void Setup()
+        public void SetUp()
         {
-            var connection = ServiceTestHelper.GetDatabaseConnection();
-            sessionService = new SessionService(connection);
-            sessionTestHelper = new SessionTestHelper(connection);
-        }
+            sessionDataService = A.Fake<ISessionDataService>();
+            A.CallTo(() => sessionDataService.StartOrRestartSession(A<int>._, A<int>._)).Returns(DefaultSessionId);
 
-        [Test]
-        public void StartOrRestartSession_Should_Start_Users_First_Session()
-        {
-            using (new TransactionScope())
-            {
-                // When
-                const int candidateId = 29;
-                const int customisationId = 100;
-                var sessionId = sessionService.StartOrRestartSession(candidateId, customisationId);
-
-                // Then
-                var newSession = sessionTestHelper.GetSession(sessionId);
-
-                newSession.Should().NotBeNull();
-                SessionTestHelper.SessionsShouldBeApproximatelyEquivalent(
-                    newSession!,
-                    SessionTestHelper.CreateDefaultSession(sessionId, candidateId, customisationId)
-                );
-            }
-        }
+            httpContextSession = new MockHttpContextSession();
 
-        [Test]
-        public void StartOrRestartSession_Should_Start_Subsequent_Session()
-        {
-            using (new TransactionScope())
-            {
-                // When
-                const int candidateId = 101;
-                const int customisationId = 1240;
-                var sessionId = sessionService.StartOrRestartSession(candidateId, customisationId);
-
-                // Then
-                var newSession = sessionTestHelper.GetSession(sessionId);
-
-                newSession.Should().NotBeNull();
-                SessionTestHelper.SessionsShouldBeApproximatelyEquivalent(
-                    newSession!,
-                    SessionTestHelper.CreateDefaultSession(sessionId)
-                );
-            }
+            sessionService = new SessionService(sessionDataService);
         }
 
         [Test]
-        public void StartOrRestartSession_Should_Stop_Other_Sessions()
+        public void Index_should_StartOrRestartSession_for_course_not_in_session()

```

Comment:
I don't think this test name matches what's being tested? 🙂

Reply:
Whoops! Fixed in 90a4212 🙂